src/rdp.c

```diff
@@ -802,6 +823,169 @@ void rdp_draw_filled_rectangle( int tx, int ty, int bx, int by )
+    __rdp_ringbuffer_queue((int)( ish * to_fixed_16_16 ) );
+    __rdp_ringbuffer_queue((int)( xm * to_fixed_16_16 ) );
+    __rdp_ringbuffer_queue((int)( ism * to_fixed_16_16 ) );
+    
+    const float mr = v2R - v1R;
+    const float mg = v2G - v1G;
+    const float mb = v2B - v1B;
+    const float hr = v3R - v1R;
+    const float hg = v3G - v1G;
+    const float hb = v3B - v1B;
+    
+    const float nxR = Hy*mr - hr*My;
+    const float nyR = hr*Mx - Hx*mr;
+    const float nxG = Hy*mg - hg*My;
+    const float nyG = hg*Mx - Hx*mg;
+    const float nxB = Hy*mb - hb*My;
+    const float nyB = hb*Mx - Hx*mb;
+    
+    const float DrDx = (fabs(nz) > FLT_MIN) ? (- nxR / nz) : 0;
+    const float DgDx = (fabs(nz) > FLT_MIN) ? (- nxG / nz) : 0;
+    const float DbDx = (fabs(nz) > FLT_MIN) ? (- nxB / nz) : 0;
+    const float DrDy = (fabs(nz) > FLT_MIN) ? (- nyR / nz) : 0;
+    const float DgDy = (fabs(nz) > FLT_MIN) ? (- nyG / nz) : 0;
+    const float DbDy = (fabs(nz) > FLT_MIN) ? (- nyB / nz) : 0;
+    
+    const float DrDe = DrDy + DrDx * ish;
+    const float DgDe = DgDy + DgDx * ish;
+    const float DbDe = DbDy + DbDx * ish;
+    
+    const int final_r = (v1R + FY * DrDe) * to_fixed_16_16;
+    const int final_g = (v1G + FY * DgDe) * to_fixed_16_16;
+    const int final_b = (v1B + FY * DbDe) * to_fixed_16_16;
+    __rdp_ringbuffer_queue( (final_r&0xffff0000) | (0xffff&(final_g>>16)) );  
+    __rdp_ringbuffer_queue( (final_b&0xffff0000) | 0x00ff ); // the 0x00ff is opaque alpha hopefully
+
+    const int DrDx_fixed = DrDx * to_fixed_16_16;
+    const int DgDx_fixed = DgDx * to_fixed_16_16;
+    const int DbDx_fixed = DbDx * to_fixed_16_16;
+
+    __rdp_ringbuffer_queue( (DrDx_fixed&0xffff0000) | (0xffff&(DgDx_fixed>>16)) );
+    __rdp_ringbuffer_queue( (DbDx_fixed&0xffff0000)  );    
+    
+    __rdp_ringbuffer_queue( 0 );  // not dealing with the color fractions right now
+    __rdp_ringbuffer_queue( 0 );
+    
+    __rdp_ringbuffer_queue( (DrDx_fixed<<16) | (DgDx_fixed&0xffff) );
+    __rdp_ringbuffer_queue( (DbDx_fixed<<16) );
+    
+    const int DrDe_fixed = DrDe * to_fixed_16_16;
+    const int DgDe_fixed = DgDe * to_fixed_16_16;
+    const int DbDe_fixed = DbDe * to_fixed_16_16;
+    
+    __rdp_ringbuffer_queue( (DrDe_fixed&0xffff0000) | (0xffff&(DgDe_fixed>>16)) );
+    __rdp_ringbuffer_queue( (DbDe_fixed&0xffff0000) );
+    
+    const int DrDy_fixed = DrDy * to_fixed_16_16;
+    const int DgDy_fixed = DgDy * to_fixed_16_16;
+    const int DbDy_fixed = DbDy * to_fixed_16_16;
+    
+    __rdp_ringbuffer_queue( (DrDy_fixed&0xffff0000) | (0xffff&(DgDy_fixed>>16)) );
+    __rdp_ringbuffer_queue( (DbDy_fixed&0xffff0000) );
+    
+    __rdp_ringbuffer_queue( (DrDe_fixed<<16) | (DgDe_fixed&0xffff) );
+    __rdp_ringbuffer_queue( (DbDe_fixed<<16) );
+    
+    __rdp_ringbuffer_queue( (DrDy_fixed<<16) | (DgDy_fixed&&0xffff) );
+    __rdp_ringbuffer_queue( (DbDy_fixed<<16) );
+    
+    __rdp_ringbuffer_send();
+    
+    return;
```

Comment:
Had a college instructor back in the day that was very insistent on it being good style and been doing it ever since. Should I take it that no explicit return from void is recommended style for libdragon? (I tried to stick to existing style, just didn't notice that element)

Reply:
Personally I think the additional `return` doesn't really add anything, so I'm not sure what makes it good style. It is a matter of preference though. But yes, I think we should stick with a consistent style in libdragon, which is to omit the `return`.